src/typing/flow_js.ml

```diff
@@ -1616,6 +1616,11 @@ let rec __flow cx (l, u) trace =
       then rec_flow cx trace (l, u)
       else rec_flow cx trace (default, u)
 
+    | (ObjTestT(reason, default, u), _) ->
```

Comment:
What is the intended behavior here? I bet object_like u is failing because u is type variable, which it will almost always be, even when defaultProps is provided (at least until the concrete object type provided as defaultProps gets unified with the type variable). Note also that l is the entire ObjTestT type here, is that what you intended in the calls below?